lib/src/conflicts.rs

```diff
@@ -437,24 +437,37 @@ pub async fn update_from_content(
     path: &RepoPath,
     content: &[u8],
 ) -> BackendResult<Merge<Option<FileId>>> {
+    let simplified_file_ids = file_ids.clone().simplify();
+    let simplified_file_ids = &simplified_file_ids;
+
     // First check if the new content is unchanged compared to the old content. If
     // it is, we don't need parse the content or write any new objects to the
     // store. This is also a way of making sure that unchanged tree/file
     // conflicts (for example) are not converted to regular files in the working
     // copy.
     let mut old_content = Vec::with_capacity(content.len());
-    let merge_hunk = extract_as_single_hunk(file_ids, store, path).await;
+    let merge_hunk = extract_as_single_hunk(simplified_file_ids, store, path).await;
     materialize_merge_result(&merge_hunk, &mut old_content).unwrap();
     if content == old_content {
         return Ok(file_ids.clone());
     }
 
-    let Some(hunks) = parse_conflict(content, file_ids.num_sides()) else {
-        // Either there are no self markers of they don't have the expected arity
+    let Some((used_file_ids, hunks)) = parse_conflict(content, simplified_file_ids.num_sides())
```

Comment:
I think this is a bit more readable:

```rust
    let (used_file_ids, hunks) = 'hunks: {
        if let Some(hunks) = parse_conflict(content, simplified_file_ids.num_sides()) {
            break 'hunks (simplified_file_ids, hunks);
        };
        if simplified_file_ids.num_sides() != file_ids.num_sides() {
            // Attempt to parse with the arity of the unsimplified conflicts.
            if let Some(hunks) = parse_conflict(content, file_ids.num_sides()) {
                break 'hunks (simplified_file_ids, hunks);
            };
        };
        // Either there are no self markers or they don't have the expected arity
        let file_id = store.write_file(path, &mut &content[..])?;
        return Ok(Merge::normal(file_id));
    };
```

A `loop {}` would also work instead of `'hunks: {}`.

Reply:
I didn't think of doing that, seems better. I was trying to figure out how to get this working with `Option` 😄